invoices/invoiceregistry.go

```diff
@@ -857,9 +860,10 @@ func (i *InvoiceRegistry) SettleHodlInvoice(preimage lntypes.Preimage) error {
 		}
 
 		i.notifyHodlSubscribers(HodlEvent{
-			CircuitKey:   key,
-			Preimage:     &preimage,
-			AcceptHeight: int32(htlc.AcceptHeight),
+			CircuitKey:       key,
+			Preimage:         &preimage,
+			AcceptHeight:     int32(htlc.AcceptHeight),
+			ResolutionResult: ResultSettled,
```

Comment:
It seems this result isn't set when `NotifyExitHop` returns a settle directly.

I think that a constructor for `HodlEvent` that forces callers to populate all fields could be useful here, to prevent subtle omissions.

Reply:
> It seems this result isn't set when NotifyExitHop returns a settle directly.

Doesn't `updateLegacy` return `SettleEvent` when `NotifyExitHopHtlc` settled?

Or does this make it look like it isn't included? Not sure if I am following this comment.

Reply:
Yes, I thought there was a case where it wasn't included. But adding the constructors will surface that, if it was already the case and not my misunderstanding.

Reply:
I could be missing it too. Going to do another check and see if I can add any more tests which check resResult then will re-request review.

Reply:
Constructors added and `ResolutionResult` checks added to all tests that call `NotifyExitHopHtlc`.